include/nbl/builtin/hlsl/projection/projection.hlsl

```diff
@@ -0,0 +1,82 @@
+#ifndef _NBL_BUILTIN_HLSL_PROJECTION_PROJECTION_INCLUDED_
+#define _NBL_BUILTIN_HLSL_PROJECTION_PROJECTION_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/concepts.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+// TODO: use glm instead for c++
+template<typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPoint<FloatingPoint>)
+inline matrix<FloatingPoint, 4, 4> buildProjectionMatrixPerspectiveFovRH(FloatingPoint fieldOfViewRadians, FloatingPoint aspectRatio, FloatingPoint zNear, FloatingPoint zFar)
```

Comment:
drop `buildProjectionMatrixPerspective` from the name, use a `nbl::hlsl::math::thin_lens` namespace instead and let the name be `rhPerspectiveFovMatrix`, likewise fo the other functions